app/assets/src/components/views/discovery/DiscoverySidebar.jsx

```diff
@@ -24,52 +14,42 @@ export default class DiscoverySidebar extends React.Component {
 
     this.state = {
       stats: {
-        samples: 0,
-        projects: 0
-        // TODO (gdingle):
-        // avg_reads_per_sample: 0,
+        samples: "",
+        projects: "",
+        total_reads: "",
+        adjusted_remaining_reads: ""
       },
       metadata: {
         host: {},
         tissue: {},
-        created_at: {}
-        // TODO (gdingle):
-        // location: {},
-      },
-      genomes: {}
+        created_at: {},
+        location: {}
+      }
     };
   }
 
-  async componentDidMount() {
-    // TODO (gdingle): include genome name in payload from getSamples
-    const genomes = keyBy("id", await getAllHostGenomes());
-    this.setState({ genomes });
-  }
-
   static getDerivedStateFromProps(newProps, prevState) {
     const { currentTab, projects } = newProps;
 
     if (currentTab == "samples") {
-      const samples = DiscoverySidebar.processSamples(
-        newProps.samples,
-        prevState.genomes
-      );
+      const samples = DiscoverySidebar.processSamples(newProps.samples);
       if (!samples || !samples.length) {
         return prevState;
       }
       return {
         stats: {
           samples: samples.length,
-          projects: uniqBy("project_id", samples).length
+          projects: uniqBy("project_id", samples).length,
```

Comment:
Rename `samples` to `numSamples`, so it's more clear that it's a count instead of an array of samples. Same with project.

Also, I know this is kind of annoying, but can you use camelCase for javascript variables? We should write something to automatically convert snake_case to camelCase at the API fetching layer so we only deal with camelCase on the front-end...

Reply:
No problem. I agree those are improvements. 

> automatically convert snake_case to camelCase at the API fetching layer 

I'd love to know whether this has been done without issue before. Personally I hate all the time I spend switching cases 😠

Reply:
Yeah it's definitely doable. The only question is whether to do it on the front-end or back-end. If we do it once the data is received on the front-end, then the data will be inconsistent with what you see in the "Network" tab on Chrome Developer tools, so back-end seems preferred. Something to look into in the future.